chain/chain/src/chain.rs

```diff
@@ -681,21 +681,19 @@ impl Chain {
                 // Set the root block of flat state to be the genesis block. Later, when we
                 // init FlatStorages, we will read the from this column in storage, so it
                 // must be set here.
-                if let Some(flat_storage_manager) = runtime_adapter.get_flat_storage_manager() {
-                    let genesis_epoch_id = genesis.header().epoch_id();
-                    let mut tmp_store_update = store_update.store().store_update();
-                    for shard_uid in
-                        epoch_manager.get_shard_layout(genesis_epoch_id)?.get_shard_uids()
-                    {
-                        flat_storage_manager.set_flat_storage_for_genesis(
-                            &mut tmp_store_update,
-                            shard_uid,
-                            genesis.hash(),
-                            genesis.header().height(),
-                        )
-                    }
-                    store_update.merge(tmp_store_update);
+                let flat_storage_manager = runtime_adapter.get_flat_storage_manager();
```

Comment:
are we not checking return value of get_flat_storage_manager() because it will always be present?

Reply:
Yes, that's the major part of this change. We always have the flat storage manager present in prod yet for some reason we were returning an option. This PR removes the option part.

Reply:
@Longarithm as far as I remember some implementations of runtime adapter do not have flat storage support, could you please confirm that?